x-pack/legacy/plugins/integrations_manager/public/data.ts

```diff
@@ -20,7 +20,9 @@ import {
   IntegrationsGroupedByStatus,
 } from '../common/types';
 
-let _fetch: HttpHandler;
+const defaultClient: HttpHandler = (path, options?) => fetch(path, options).then(res => res.json());
```

Comment:
Use window.fetch / `whatwg-fetch` which returns parsed JSON response as default client. Reasonable default (very similar to `kfetch`/`core.http.fetch`) and prevents error if missing (e.g. `setClient` isn't called first)